File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    answer = 0
    sub = 100
    array.sort()
    
    for i, x in enumerate(array):
        if abs(x-n) < sub or (abs(x-n) == sub and array[i] < answer):
            sub = abs(x-n)
            answer = x
            
    return answer

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = 0;
        int tmp = 100;
        int x = 0;
        
        for (int i=0; i<array.length; i++) {
            int sub = Math.abs(array[i]-n);
            if (sub < tmp || (sub == tmp && array[i] < array[x])) {
                tmp = sub;
                answer = array[i];
                x = i; 
            }
        }
        
        return answer;
    }
}